### PR TITLE
Update mtuhandler to support platform running nftables

### PR DESCRIPTION
**Pull request by yboaron**
Submariner MTU driver in route-agent programs iptable rules to
clamp tcp mss to PMTU in the Mangle table. When the underlying
platform is running nftables, the auto-translate (iptables-translate)
 fails to convert these rules (most probably because
iptables-translate currently doesn't support converting mss-clamping
rules targeting ipsets).

This PR configures mss-clamping rules that don't target ipsets when
the underlying platform is running nftables.

Once auto-translate for mss-clamping rules targeting ipsets is
supported, this fix could be reverted.

Signed-off-by: yboaron <yboaron@redhat.com>

### CODE REVIEW

File: pkg/iptables/iptables.go
Comment (tpantelis) [inline pkg/iptables/iptables.go]: These don't need to be publicly scoped. That said they're only used in one place and thus don't need to be constants.

File: pkg/iptables/iptables.go
Comment (tpantelis) [inline pkg/iptables/iptables.go:202]: This function doesn't use `ipt`.  It seems this function should be part of the `Interface`?

File: pkg/iptables/iptables.go
Comment (tpantelis) [inline pkg/iptables/iptables.go]:
```suggestion
	return strings.Contains(out.String(), NftSubstr)
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/routeagent_driver/handlers/mtu/mtuhandler.go
Comment (sridhargaddam) [inline pkg/routeagent_driver/handlers/mtu/mtuhandler.go:114]: Before we merge this PR, let us confirm that its indeed working on a real setup.

Here we are avoiding adding rules with localCIDR, so I'd like to hear from @aswinsuryan if this is fine and if there were any other reasons why we used to add iptable rules even with localCIDRs.

File: pkg/iptables/iptables.go
Comment (sridhargaddam) [inline pkg/iptables/iptables.go]:
```suggestion
		klog.Warningf("Unable to run %s %s: %q", IptablesCmd, VersionCmd, err)
```

File: pkg/iptables/iptables.go
Comment (sridhargaddam) [reply pkg/iptables/iptables.go]: +1 to "dont need to be publicly scoped"

The info messages below seem to use hardcoded names, so we can instead use these macros.

File: pkg/iptables/iptables.go
Comment (sridhargaddam) [inline pkg/iptables/iptables.go]:
```suggestion
		klog.Warningf("Unable to find %s path: %q", IptablesCmd, err)
```

**Review by sridhargaddam — CHANGES_REQUESTED** [review]
Thanks for the PR @yboaron. Some minor comments. Adding "Requested changes" mainly to hold this PR from getting merged until we validate the fix on an OSP/AWS setup.

File: pkg/iptables/iptables.go
Comment (yboaron) [reply pkg/iptables/iptables.go:202]: Yes you are right, I thought to include it in the interface to make it easier to retrieve mode from iptables library when supported.

Both options seem fine to me, would you rather exclude it from the interface?

File: pkg/routeagent_driver/handlers/mtu/mtuhandler.go
Comment (yboaron) [reply pkg/routeagent_driver/handlers/mtu/mtuhandler.go:114]: Yep, agree, we need to verify it solves the mtu issue with nftables platforms, before merging it.

Comment (aswinsuryan) [reply pkg/routeagent_driver/handlers/mtu/mtuhandler.go:114]: It was added since some tests were failing, IIRC some of the host networking tests. If e2e passes it should be safe to avoid these rules.

File: pkg/iptables/iptables.go
Comment (tpantelis) [reply pkg/iptables/iptables.go:202]: I'd prefer on the interface so we can mock it in unit tests.

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Comment by yboaron** [comment]
I'm going to close this PR , and will open new PR to address MTU issue

**Pull request closed without merging**
